### PR TITLE
Refactor Solr::Testing to not use instrumentation

### PR DESCRIPTION
**Pull request by drnluz**

### CODE REVIEW

File: lib/solr/testing.rb [hunk 1]
```diff
@@ -1,19 +1,19 @@
...
+module Solr::Query::Request::RunnerExtension
+  def run
+    response = super
+    Solr::Testing.last_solr_request_params = request_params[:params]
+    response
+  end
+end
+
+class Solr::Query::Request::Runner
+  prepend Solr::Query::Request::RunnerExtension
```
Comment (Valve) [inline lib/solr/testing.rb]: I didn't know about prepend, what does it do?

Comment (Valve) [reply lib/solr/testing.rb]: can you please update the readme with an example of how to use it?

Comment (drnluz) [reply lib/solr/testing.rb]: The difference between `prepend` and `include` is that `prepend` will add the module before the original class on the ancestors chain.
The basic idea is to include my extension in a way that the extension's `run` method is found by ruby before the original `run` method. This allows me to still access the original method using `super` and do the additional stuff (storing request params for example) without having to duplicate the original method

Sure, I'll add a section to the readme.

Comment (Valve) [reply lib/solr/testing.rb]: that's cool, thanks

Comment (drnluz) [reply lib/solr/testing.rb]: this post explains better: http://leohetsch.com/include-vs-prepend-vs-extend/

Comment (SeTeM) [reply lib/solr/testing.rb]: @valve, I’ve already used perpend on machinio, as I remember

**Comment by drnluz** [comment]
@Valve Added a section to readme, let me know what you think

File: README.md [hunk 2]
```diff
@@ -305,6 +306,24 @@ ActiveSupport::Notifications.subscribe('request.solrb')  do |*args|
 end
 ```
 
+# Testing
+
+It's possible to inspect the parameters for each solr query request done using Solrb by requiring
+`solr/testing` file on your test suite. The query parameters will be accessible by reading
```
Comment (Valve) [inline README.md]: by requiring ... *in* your test suite?

Comment (drnluz) [reply README.md]: thanks, fixed

**Pull request merged**